nomos-services/key-management-system/src/lib.rs

```diff
@@ -0,0 +1,210 @@
+use crate::backend::KMSBackend;
+use crate::secure_key::SecuredKey;
+use bytes::Bytes;
+use either::Either;
+use futures::StreamExt;
+use log::error;
+use overwatch_rs::services::handle::ServiceStateHandle;
+use overwatch_rs::services::life_cycle::LifecycleMessage;
+use overwatch_rs::services::relay::RelayMessage;
+use overwatch_rs::services::state::{NoOperator, NoState};
+use overwatch_rs::services::{ServiceCore, ServiceData, ServiceId};
+use overwatch_rs::DynError;
+use std::fmt::{Debug, Formatter};
+use std::future::Future;
+use tokio::sync::oneshot;
+
+mod backend;
+mod secure_key;
+
+const KMS_TAG: ServiceId = "KMS";
+
+// TODO: Use [`AsyncFnMut`](https://doc.rust-lang.org/stable/std/ops/trait.AsyncFnMut.html#tymethod.async_call_mut) once it is stabilized.
+pub type KMSOperator = Box<
+    dyn FnMut(&mut dyn SecuredKey) -> Box<dyn Future<Output = Result<(), DynError>>> + Send + Sync,
```

Comment:
I don't clearly understand how the caller of the `KMSMessage::Execute` receives an output. Do we probably need to add a `oneshot::Sender` to this `Fn`?

Reply:
Ah, right. That totally makes sense. Thanks. I'm resolving this comment.